src/fmu/dataio/_model/global_configuration.py

```diff
@@ -137,6 +137,36 @@ def __iter__(self) -> Any:
     def __getitem__(self, item: str) -> StratigraphyElement:
         return self.root[item]
 
+    def _get_index(self, name: str) -> int:
+        """
+        Get the index of a stratigraphic element by name.
+        """
+        return list(self.root.keys()).index(name)
+
+    def get_official_name(self, name: str) -> str:
+        """
+        Get the official SMDA name of a stratigraphic element.
+        name: name of stratigraphic element
+        """
+        if name in self.root:
+            return self.root[name].name
+        return ""
+
+    def get_ordered_official_names(self, names: list[str]) -> list[str]:
+        """
+        Retrieve the official SMDA names and order them according to
+        the stratigraphic column.
+        names: names of stratigraphic elements
+        """
+        try:
+            sorted_indices = sorted([self._get_index(name) for name in names])
+            return [
+                self.get_official_name(list(self.root.keys())[idx])
+                for idx in sorted_indices
+            ]
+        except ValueError:
+            return []
```

Comment:
I guess this is catching if `_get_index()` raises; if one entry does not exist, should we abort on _all_ entries and return the empty list, or just ignore that one and carry on with the rest?